documentation/src/main/asciidoc/topics/infinispan_testcontainers.adoc

```diff
@@ -0,0 +1,44 @@
+# Testing using {brandname} with Testcontainers
+
+This module helps running [{brandname} Server](https://infinispan.org) using
+[Testcontainers](https://www.testcontainers.org/).
+
+It is based on the [official container image](https://github.com/infinispan/infinispan-images/) provided by Infinispan.
+
+## Usage example
+
+You can start the {brandname} container instance from any Java application:
+
+[source,java]
+----
+include::code_examples/InfinispanContainer.java[]
+----
+
+
+## Choose credentials
+
+The Infinispan server is configured with the following credentials by default:
+
+* username: `Titus Bramble`
+* password: `Shambles`
+
+You can change them by using the `withUser()` and `withPassword()`
+methods before starting the container:
+
+[source,java]
+----
+include::code_examples/InfinispanContainerCredentials.java[]
+----
+
+## Adding the testcontainer to your project dependencies
+
+Add the following dependency to your `pom.xml` file:
+
+```xml tab='Maven'
+<dependency>
+    <groupId>org.infinispan</groupId>
+    <artifactId>infinispan-server-testdriver-core</artifactId>
+    <version>{{latest_version}}</version>
+    <scope>test</scope>
+</dependency>
+```
```

Comment:
```suggestion
= Testing {brandname} with test containers

Run link:https://infinispan.org[{brandname} Server] instances using link:https://www.testcontainers.org[Testcontainers] for faster, simpler JUnit tests.

The {brandname} Server test container is based on the link:https://github.com/infinispan/infinispan-images/[official container image] provided by {brandname}.

== Starting the {brandname} test container

You can start the {brandname} Server container instance from any Java application as follows:

[source,java]
----
include::code_examples/InfinispanContainer.java[]
----

== Changing credentials

{brandname} Server is configured with the following default credentials:

* Username: `Titus Bramble`
* Password: `Shambles`

Add custom credentials with the `withUser()` and `withPassword()` methods before starting the container:

[source,java]
----
include::code_examples/InfinispanContainerCredentials.java[]
----

== Adding the test container to your project dependencies

Add the following dependency to your `pom.xml` file:

[source,xml,options="nowrap"]
----
<dependency>
    <groupId>org.infinispan</groupId>
    <artifactId>infinispan-server-testdriver-core</artifactId>
    <version>{{latest_version}}</version>
    <scope>test</scope>
</dependency>
----
```

Reply:
.md to .adoc and a few suggestions. 

@tristantarrant I guess we can figure out the downstream stuff later, but would it make sense to create a "Testing Infinispan" assembly? I was just looking at where we could reference this in the Server Guide but I'm not sure if that would be overkill.

Reply:
This is more for a developer guide, rather than the server guide. "Testing with Infinispan" would be a more appropriate title